lib/elixir/src/elixir_parser.yrl

```diff
@@ -294,19 +294,23 @@ stab -> stab eoe stab_expr : ['$3'|'$1'].
 stab_eoe -> stab : '$1'.
 stab_eoe -> stab eoe : '$1'.
 
-stab_expr -> expr : '$1'.
-stab_expr -> stab_op_eol stab_maybe_expr : build_op('$1', [], '$2').
-stab_expr -> empty_paren stab_op_eol stab_maybe_expr :
-               build_op('$2', [], '$3').
-stab_expr -> call_args_no_parens_all stab_op_eol stab_maybe_expr :
-               build_op('$2', unwrap_when(unwrap_splice('$1')), '$3').
-stab_expr -> stab_parens_many stab_op_eol stab_maybe_expr :
-               build_op('$2', unwrap_splice('$1'), '$3').
-stab_expr -> stab_parens_many when_op expr stab_op_eol stab_maybe_expr :
-               build_op('$4', [{'when', meta_from_token('$2'), unwrap_splice('$1') ++ ['$3']}], '$5').
-
-stab_maybe_expr -> 'expr' : '$1'.
-stab_maybe_expr -> '$empty' : nil.
+%% Here, `element(1, Token)` is the stab operator, while `element(2, Token)` is
+%% the expression.
+stab_expr -> expr :
+               '$1'.
+stab_expr -> stab_op_eol_and_maybe_expr :
+               build_op(element(1, '$1'), [], element(2, '$1')).
+stab_expr -> empty_paren stab_op_eol_and_maybe_expr :
+               build_op(element(1, '$2'), [], element(2, '$2')).
+stab_expr -> call_args_no_parens_all stab_op_eol_and_maybe_expr :
+               build_op(element(1, '$2'), unwrap_when(unwrap_splice('$1')), element(2, '$2')).
+stab_expr -> stab_parens_many stab_op_eol_and_maybe_expr :
+               build_op(element(1, '$2'), unwrap_splice('$1'), element(2, '$2')).
+stab_expr -> stab_parens_many when_op expr stab_op_eol_and_maybe_expr :
+               build_op(element(1, '$4'), [{'when', meta_from_token('$2'), unwrap_splice('$1') ++ ['$3']}], element(2, '$4')).
+
+stab_op_eol_and_maybe_expr -> stab_op_eol expr : {'$1', '$2'}.
```

Comment:
I added `maybe` to the name to remain consistent to how it was called before (`stab_maybe_expr`) and also because we're not sure there's an expression there. You're suggesting `stab_op_eol_and_expr`?